ui/src/app/pages/admin/admin-lock-request.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {
+  SemiBoldHeader
+} from 'app/components/headers';
+import {DatePicker, TextArea} from 'app/components/inputs';
+import {Button} from 'app/components/buttons';
+import {workspaceAdminApi} from 'app/services/swagger-fetch-clients';
+import {useState} from 'react';
+import {TooltipTrigger} from 'app/components/popups';
+import colors from 'app/styles/colors';
+
+interface Props {
+  workspace: string,
+  onLock: Function,
+  onClose: Function
+}
+
+export const AdminLockRequest = (props: Props) => {
+  const [requestReason, setRequestReason] = useState('');
+  const [requestDate, setRequestDate] = useState(new Date());
+  const [showError, setShowError] = useState(false);
+
+
+  const disableLockWorkspace = !requestReason ||
+      requestReason === '' || !requestDate || requestDate?.toString() === '' || showError;
+
+  const getToolTipContent = showError ? 'Error occurred while Locking Workspace' :
+      'Request Reason & Request Date are required to lock workspace';
+
+  const onLockWorkspace = () => {
+    const {workspace , onClose, onLock} = props;
+    const adminLockingRequest = {
+      requestReason,
+      requestDateInMillis: requestDate.valueOf()
+    }
+
+    workspaceAdminApi().setAdminLockedState(workspace, adminLockingRequest)
+      .then(() => {
+        onLock();
+        onClose();
```

Comment:
not sure we want to be calling `onClose()` here.

By do so, we require the `onLock()` to be something that's appropriate to run for both locking and closing, and that may not always be true.

Reply:
Good point